File: Design-Patterns/Visitor/Cpp/equipment.cpp
#include "equipment.h"

equipment::equipment(std::string name, int loading)
{
	name_ = name;
	loading_ = loading;
}

equipment::~equipment() = default;

std::string equipment::get_name()
{
	return name_;
}

int equipment::get_loading()
{
	return loading_;
}

File: Design-Patterns/Visitor/CSharp/Equipments/Equipment.cs
using Visitor.Visitors;

namespace Visitor.Equipments
{
    public abstract class Equipment
    {
        public string Name { get; protected set; }
        public int Loading { get; protected set; }

        protected Equipment(string name, int loading)
        {
            Name = name;
            Loading = loading;
        }

        public abstract void RunVisitor(EquipmentVisitor visitor);
    }
}
